### PR TITLE
Set proxy vars into default file in /etc/default/rke2-server or agent

### PR DESCRIPTION
**Pull request by santhoshdaivajna**
set containerd proxy vars
update kairos to latest

### CODE REVIEW

**Review by rishi-anand — APPROVED** [review]

File: main.go [hunk 1]
```diff
@@ -2,21 +2,27 @@ package main
...
 	"fmt"
-	"path/filepath"
-
-	"github.com/c3os-io/c3os/sdk/clusterplugin"
+	"github.com/kairos-io/kairos/pkg/config"
+	"github.com/kairos-io/kairos/sdk/clusterplugin"
 	yip "github.com/mudler/yip/pkg/schema"
 	"github.com/sirupsen/logrus"
 	"gopkg.in/yaml.v2"
+	"net"
```
Comment (rosskirkpat) [inline main.go:12]: goimports

**Review by rosskirkpat — CHANGES_REQUESTED** [review]

File: main.go [hunk 2]
```diff
@@ -96,6 +121,68 @@ func clusterProvider(cluster clusterplugin.Cluster) yip.YipConfig {
 	return cfg
 }
 
+func proxyEnv(userOptions []byte) string {
+	var proxy []string
+
+	httpProxy := os.Getenv("HTTP_PROXY")
```
[the remark this answers is not in the document]
Comment (santhoshdaivajna) [reply main.go:127]: @rosskirkpat  what do you mean by -> "in the implementation that consumes the provider."

the rke2-server and the agent read config info from /etc/default/rke2-server or agent files while starting, so the file is being set with the respective proxy variables.

File: main.go [hunk 3]
```diff
@@ -25,6 +31,8 @@ type RKE2Config struct {
 	TLSSan []string `yaml:"tls-san"`
 }
 
+var configScanDir = []string{"/oem", "/usr/local/cloud-config", "/run/initramfs/live"}
```
Comment (paynejacob) [inline main.go:34]: cluster provider plugins should not read configuration directly the configuration will be passed by kairos.

**Review by paynejacob — CHANGES_REQUESTED** [review]

File: main.go (hunk 3 above)
Comment (santhoshdaivajna) [reply main.go:35]: the cluster object in the provider does not contain the env config values.

**Review by rishi-anand — APPROVED** [review]

**Review by paynejacob — CHANGES_REQUESTED** [review]
we should not be loading any configuration in the providers, this breaks the contract for cluster providers in kairos.  We should configure rke2 to read from /etc/environment rather than try to generate this configuration in the provider.

**Comment by santhoshdaivajna** [comment]
I will file this as an improvement and pick this up for all the providers (rke2/k3s/kubeadm) separately.

**Review by paynejacob — APPROVED** [review]

**Pull request merged**